src/cryo_challenge/utils/_fourier_statistics.py

```diff
@@ -0,0 +1,314 @@
+        If `"linear"`, evaluate the grid using linear
+        interpolation. If `"nearest"`, use nearest-neighbor
+        interpolation.
+    - `minimum_frequency`:
+        The minimum frequency to include in the power spectrum.
+    - `maximum_frequency`:
+        The maximum frequency to include in the power spectrum.
+    - `chunk_size`:
+        The chunk size used for vmapped operations.
+    **Returns:**
+    - `radially_averaged_powerspectrum_on_grid`:
+        The average of the radially averaged power spectrums of the volumes in the stack
+        interpolated onto a radial frequency grid.
+    """
+
+    radially_averaged_powerspectrum, radial_frequency_grid, frequency_bins = (
+        _compute_averaged_powerspectrum_from_stack(
+            volume_stack,
+            voxel_size=voxel_size,
+            minimum_frequency=minimum_frequency,
+            maximum_frequency=maximum_frequency,
+            chunk_size=chunk_size,
+        )
+    )
+    radially_averaged_powerspectrum_on_grid = _interpolate_radial_average_on_grid(
+        radially_averaged_powerspectrum,
+        frequency_bins,
+        radial_frequency_grid,
+        interpolation_mode=interpolation_mode,
+    )
+
+    return radially_averaged_powerspectrum_on_grid
+
+
+def compute_radially_averaged_powerspectrum(
+    fourier_volume: Tensor,
+    radial_frequency_grid: Tensor,
+    voxel_size: float = 1.0,
+    *,
+    minimum_frequency: float = 0.0,
+    maximum_frequency: float = math.sqrt(2) / 2,
+) -> Tensor:
+    """
+    Compute the radially averaged power spectrum of a volume in fourier space.
+
+    **Arguments:**
+    - `fourier_volume`:
+        A volume in fourier space, such as the output of `cryo_challenge.fft.rfftn`.
+    - `radial_frequency_grid`:
+        The radial frequency coordinate system of the fourier_volume.
+    - `voxel_size`:
+        The voxel size of the volume. If not provided, the power spectrum
+        is computed in grid units.
+    - `minimum_frequency`:
+        The minimum frequency to include in the power spectrum.
+    - `maximum_frequency`:
+        The maximum frequency to include in the power spectrum.
+
+    **Returns:**
+    - `radially_averaged_powerspectrum`:
+        The radially averaged power spectrum of the volume.
+    - `frequency_bins`:
+        The array of frequencies for which we have calculated the
+        radially averaged power spectrum.
+    """
+    squared_fourier_amplitudes = (fourier_volume * fourier_volume.conj()).real
+    frequency_bins = _make_radial_frequency_bins(
+        fourier_volume.shape,
+        minimum_frequency,
+        maximum_frequency,
+        voxel_size,
+    )
+
+    radially_averaged_powerspectrum = _compute_binned_radial_average(
+        squared_fourier_amplitudes, radial_frequency_grid, frequency_bins
+    )
+
+    return radially_averaged_powerspectrum, frequency_bins
+
+
+def compute_fourier_shell_correlation(
+    fourier_volume_1: Tensor,
+    fourier_volume_2: Tensor,
```

Comment:
yes, good to factor out the fourier transform, so does not have to be recomputed